src/calendarworker.cpp

```diff
@@ -221,18 +242,18 @@ bool CalendarWorker::sendResponse(const QString &uid, const QDateTime &recurrenc
         if (!sent)
             updateAttendee(event, updated, origAttendee);
     } else {
-        qWarning() << "Failed to load notebook for incidence" << event->instanceIdentifier();
+        qWarning() << "Failed to load notebook for incidence" << instanceId;
     }
 
     return sent;
 }
 
-QString CalendarWorker::convertEventToICalendar(const QString &uid, const QString &prodId) const
+QString CalendarWorker::convertEventToICalendar(const QString &instanceId, const QString &prodId) const
 {
     // NOTE: not fetching eventInstances() with different recurrenceId
-    KCalendarCore::Event::Ptr event = mCalendar->event(uid);
+    KCalendarCore::Incidence::Ptr event = mCalendar->instance(instanceId);
```

Comment:
Earlier fetched the main event by just using uid. Should that be taken into account, converting instance with recurrence id? Should fetch the parent if recurring?

Reply:
I'm happy you raise this point. I noticed it when making the changes.

In the previous code, if you export an exception from the UI, it will actually create a VEVENT for the parent only. There is no appending of the exceptions or what ever. To keep it simple, I decided to change to actually export the exception only in that case.

It's maybe not the best idea I had, and I would understand if you think it's better not to change any behaviour with this commit and possibly address it in a later dedicated PR.